### PR TITLE
Added compiler directive to expose JsonNavigator and CosmosElement in internal builds

### PR DESCRIPTION
**Pull request by john-pao**
# Pull Request Template

## Description

Added #if INTERNAL compiler directive to change JsonNavigator and CosmosElement accessibility level to public in internal builds. Having direct access to these classes is a requirement for Gremlin API moving to the V3 SDK.

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/CosmosElements/CosmosArray.EagerCosmosArray.cs [hunk 1]
```diff
@@ -7,7 +7,12 @@ namespace Microsoft.Azure.Cosmos.CosmosElements
     using System.Collections.Generic;
     using Microsoft.Azure.Cosmos.Json;
 
+#if INTERNAL
+#pragma warning disable SA1601 // Partial elements should be documented
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/CosmosElements/CosmosArray.EagerCosmosArray.cs]: Please limits #if to accessor public/internal only.

**Comment by kirankumarkolli** [comment]
Do all these API's needed? My understanding is just the top level CosmosElement types are enough. 
its an very extensive contract for dependency.

File: Microsoft.Azure.Cosmos/src/CosmosElements/CosmosArray.EagerCosmosArray.cs [hunk 2]
```diff
@@ -7,7 +7,12 @@ namespace Microsoft.Azure.Cosmos.CosmosElements
...
+    public abstract partial class CosmosArray : CosmosElement, IReadOnlyList<CosmosElement>
```
[the remark this answers is not in the document]
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/CosmosElements/CosmosArray.EagerCosmosArray.cs]: Is it possible to do the entire namespace instead?

**Comment by john-pao** [comment]
> Contract surface is too big for INTERNAL dependency.

> Do all these API's needed? My understanding is just the top level CosmosElement types are enough.
> its an very extensive contract for dependency.

There may be a few cases in which I can reduce some of the scope. However, in any given set of files containing a partial class (e.g. CosmosString.cs, CosmosString.EagerCosmosString.cs, and CosmosString.LazyCosmosString.cs) I need to maintain the same accessibility modifier on ALL instances of the partial class CosmosString. I am not exposing the private implementations such as EagerCosmosString or LazyCosmosString.

File: Microsoft.Azure.Cosmos/src/CosmosElements/CosmosArray.EagerCosmosArray.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (john-pao) [reply Microsoft.Azure.Cosmos/src/CosmosElements/CosmosArray.EagerCosmosArray.cs]: > Is it possible to do the entire namespace instead?

I'm not certain what you mean here. Include the namespace in the #if def?

**Comment by kirankumarkolli** [comment]
Please create internal/friends UT to cover the usage scenarios. 
SDK can guarantee only those in the UT.

**Pull request merged**